pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -100,7 +100,8 @@ func (b *AutoscalingGroupModelBuilder) Build(c *fi.ModelBuilderContext) error {
 				warmPoolTask.MaxSize = warmPool.MaxSize
 
 				if warmPool.EnableLifecycleHook {
-					name := "kops-warmpool"
+					hookName := "kops-warmpool"
+					name := fmt.Sprintf("%s-%s", hookName, ig.GetName())
 
 					lifecyleTask := &awstasks.AutoscalingLifecycleHook{
```

Comment:
should HookName be passed in here?

Reply:
yeah just caught this one. think we need some integration tests for this :)